<a href="https://colab.research.google.com/github/JSJeong-me/KOSA-Big-Data_Vision/blob/main/Model/brownlee-chp8-chi-classif.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
# Feature Extraction with Univariate Statistical Tests (Chi-squared for classification)
from pandas import read_csv
from numpy import set_printoptions
from sklearn.feature_selection import SelectKBest
from sklearn.feature_selection import chi2
from sklearn.feature_selection import f_classif


In [2]:
# load data
filename = 'pima-indians-diabetes.data.csv'
names = ['preg', 'plas', 'pres', 'skin', 'test', 'mass', 'pedi', 'age', 'class']

In [3]:
dataframe = read_csv(filename, names=names)


In [ ]:
dataframe.describe()

In [5]:
array = dataframe.values
X = array[:,0:8]
Y = array[:,8]
# feature extraction
test = SelectKBest(score_func=f_classif, k=4)


In [6]:
fit = test.fit(X, Y)
# summarize scores
set_printoptions(precision=3)
print(fit.scores_)
features = fit.transform(X)
# summarize selected feat

[ 39.67  213.162   3.257   4.304  13.281  71.772  23.871  46.141]


f_classif :[ (1)39.67  (2)213.162   (3)3.257   (4)4.304  (5)13.281  (6)71.772  (7)23.871  (8)46.141] -> selected (2) (6) (8) (1)


chi-square: [ (1)111.52  (2)1411.887   (3)17.605   (4)53.108 (5)2175.565  (6)127.669   (7) 5.393  (8)181.304] ->  (5) (2) (8) (6
)

dataframe = read_csv(filename, names=names)
array = dataframe.values
X = array[:,0:8]
Y = array[:,8]
# feature extraction
test = SelectKBest(score_func=chi2, k=4)
fit = test.fit(X, Y)
# summarize scores
set_printoptions(precision=3)
print(fit.scores_)
features = fit.transform(X)
# summarize selected feat

In [ ]:
dataframe.shape

(768, 9)

In [ ]:
features.shape

(768, 4)

In [ ]:
print(features[0:5,:])


[[148.    0.   33.6  50. ]
 [ 85.    0.   26.6  31. ]
 [183.    0.   23.3  32. ]
 [ 89.   94.   28.1  21. ]
 [137.  168.   43.1  33. ]]
